### PR TITLE
Add semantic conventions for OS process as resource attributes

### LINKED ISSUE #891: Add semantic conventions for process resource to `api/standard` package
| Attribute  | Description  | Example  | Required |
|---|---|---|--|
| process.pid | Process identifier (PID). | `1234` | Yes |
| process.executable.name | The name of the process executable. On Unix based systems, can be set to the `Name` in `proc/[pid]/status`. On Windows, can be set to the base name of `GetProcessImageFileNameW`. | `otelcol` | See below |
| process.executable.path | The full path to the process executable. On Unix based systems, can be set to the target of `proc/[pid]/exe`. On Windows, can be set to the result of `GetProcessImageFileNameW`. | `/usr/bin/cmd/otelcol` | See below |
| process.command | The command used to launch the process (i.e. the command name). On Unix based systems, can be set to the zeroth string in `proc/[pid]/cmdline`. On Windows, can be set to the first parameter extracted from `GetCommandLineW`. | `cmd/otelcol` | See below |
| process.command_line | The full command used to launch the process. The value can be either a list of strings representing the ordered list of arguments, or a single string representing the full command. On Unix based systems, can be set to the list of null-delimited strings extracted from `proc/[pid]/cmdline`. On Windows, can be set to the result of `GetCommandLineW`. | Unix: `[ cmd/otecol, --config=config.yaml ]`, Windows: `cmd/otecol --config=config.yaml` | See below |
| process.owner | The username of the user that owns the process. | `root` | No |

Related to https://github.com/open-telemetry/opentelemetry-specification/pull/635

### PR DESCRIPTION
**Pull request by MrAlias**
Resolves #891

### CODE REVIEW

File: api/standard/resource.go [hunk 1]
```diff
@@ -92,6 +92,34 @@ const (
 	FaaSInstance = kv.Key("faas.instance")
 )
 
+// Standard operation system process resource attribute keys.
```
Comment (evantorrie) [inline api/standard/resource.go:95]: I think this is meant to be

```suggestion
// Standard operating system process resource attribute keys.
```

File: api/standard/resource.go [hunk 2]
```diff
@@ -92,6 +92,34 @@ const (
...
+const (
+	// Process identifier (PID).
+	ProcessPIDKey = kv.Key("process.pid")
+	// The name of the process executable. On Unix based systems, can be set
+	// to the `Name` in `proc/[pid]/status`. On Windows, can be set to the
+	// base name of `GetProcessImageFileNameW`.
+	ProcessExecutableNameKey = kv.Key("process.executable.name")
+	// The full oath to the process executable. On Unix based systems, can
```
Comment (evantorrie) [inline api/standard/resource.go:103]:
```suggestion
	// The full path to the process executable. On Unix based systems, can
```

File: api/standard/resource.go [hunk 3]
```diff
@@ -92,6 +92,34 @@ const (
...
+	// `proc/[pid]/cmdline`. On Windows, can be set to the first parameter
+	// extracted from `GetCommandLineW`.
+	ProcessCommandKey = kv.Key("process.command")
+	// The full command used to launch the process. The value can be either
+	// a list of strings representing the ordered list of arguments, or a
+	// single string representing the full command. On Unix based systems,
+	// can be set to the list of null-delimited strings extracted from
+	// `proc/[pid]/cmdline`. On Windows, can be set to the result of
+	// `GetCommandLineW`.
+	ProcessCommandLineKey = kv.Key("process.command_line")
```
Comment (evantorrie) [inline api/standard/resource.go:118]: Perhaps all of the above comments about using `proc/[pid]/...` should be changed to say `On Linux based systems,`.

If I recall, on FreeBSD, `procfs` is not enabled by default, and on Darwin, it just plain doesn't exist.

Comment (evantorrie) [reply api/standard/resource.go:118]: Ah, I see that this comments came straight from the specification. Well, one may say the specification is a little too Linux specific I guess.

File: api/standard/resource.go (hunk 1 above)
Comment (MrAlias) [reply api/standard/resource.go]: whoops, thanks :+1:

File: api/standard/resource.go (hunk 2 above)
Comment (MrAlias) [reply api/standard/resource.go]: thanks again :)

File: api/standard/resource.go (hunk 3 above)
Comment (MrAlias) [reply api/standard/resource.go:118]: Created https://github.com/open-telemetry/opentelemetry-specification/pull/695, will update here.

Comment (MrAlias) [reply api/standard/resource.go:118]: Updated.

**Pull request merged**